### ISSUE TITLE
The actual password used for encryption consists mostly from stars

### ISSUE THREAD

**Issue by p-an** [TeamWin/Team-Win-Recovery-Project#661]
Hi,
the actual password used for backup encryption consists almost only from stars (*), only the last character is from the password typed in - e.g. if I type abcd as password, the string passed to openaes is ***d. This makes all the the passwords weak, since there are (if I am not wrong) only 32*256 = 8192 combinations (the unknowns are only the positions of the last character and it value), practically even less, because not all of 256 byte values  can be written on the keyboard.

See the output of ps commnad during backup. At first I thought that it is some kind of protection, so  the password is not visible in this list, but it is the value, which is actually used.

Thank you

Pavel

**Comment by viisauksena** [comment]
you mean google password for phone encryption is so weak, can you point to the source where password function is

**Comment by p-an** [comment]
No, I mean the password used for encrypting the backup which is created by TWRP. The password that is passed to openaes. The password I type when tap on backup in TWPR and then on Encryption. Whatever I type, all the letters from the password except the last one are replaced with stars, not only on the screen, but strings which is passed to openaes is modified in this way. I found this when I was trying to extract the encrypted backup on my PC. The password I typed in TWRP didn`t work and after some debugging (I added some fprintfs into openaes to see the actual key) I discovered what is going on.
